src/lib.rs

```diff
@@ -1,7 +1,33 @@
+use std::{borrow::Cow, io};
+use thiserror::Error;
+
 #[macro_use]
 mod macros;
 
 #[cfg(feature = "image")]
 pub mod image;
 #[cfg(feature = "runtime")]
 pub mod runtime;
+
+pub type Result<T> = std::result::Result<T, OciSpecError>;
+
+#[derive(Error, Debug)]
+pub enum OciSpecError {
+    #[error("{0}")]
+    General(String),
+    #[error("io operation failed")]
+    Io(#[from] io::Error),
+    #[error("serde failed")]
+    SerDe(#[from] serde_json::Error),
+}
+
+fn oci_error<'a, M>(message: M) -> OciSpecError
+where
+    M: Into<Cow<'a, str>>,
+{
+    let message = message.into();
+    match message {
+        Cow::Borrowed(s) => OciSpecError::General(s.to_owned()),
+        Cow::Owned(s) => OciSpecError::General(s),
+    }
+}
```

Comment:
Do we wanna put this into a new `error` module?

Reply:
Sure, why not.